es5-shim.js

```diff
@@ -1408,14 +1408,16 @@ var hasTrimWhitespaceBug = StringPrototype.trim && (ws.trim() || !zeroWidth.trim
 defineProperties(StringPrototype, {
     // http://blog.stevenlevithan.com/archives/faster-trim-javascript
     // http://perfectionkills.com/whitespace-deviations/
+    // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/trim
     trim: function trim() {
-        if (typeof this === 'undefined' || this === null) {
+        // Make sure we trim BOM and NBSP
+        var rtrim = new RegExp('^'+wsRegexChars+'+|'+wsRegexChars+'+$','g');
```

Comment:
This regex should be created outside the function, so it's not created every time. See `trimBeginRegexp` and `trimEndRegExp`, which should be removed if they're not used anymore.